pkg/reconciler/revision/resources/deploy.go

```diff
@@ -112,6 +134,20 @@ func makePodSpec(rev *v1.Revision, cfg *config.Config) (*corev1.PodSpec, error)
 		}
 	}
 
+	// If concurrencyStateEndpoint is enabled, add the serviceAccountToken to QP via a projected volume
+	if cfg.Deployment.ConcurrencyStateEndpoint != "" {
+		podSpec.Volumes = append(podSpec.Volumes, varProjectedVolume)
+
+		for i, container := range podSpec.Containers {
+			if container.Name == QueueContainerName {
+				varProjectedMount := varProjectedVolumeMount.DeepCopy()
+				container.VolumeMounts = append(container.VolumeMounts, *varProjectedMount)
+
+				podSpec.Containers[i] = container
```

Comment:
Instead of this, I'd do

```golang
for i := range podSpec.Containers {
  container := &podSpec.Containers[i]
  ...
}
```

which'll give you a reference to the container, so you can change it in-place.